doc/cygnus-common/installation_and_administration_guide/management_interface_v1.md

```diff
@@ -929,3 +933,105 @@ When an invalid `transient` parameter is given:
 ```
 
 [Top](#top)
+
+##<a name="section7"></a>Metrics
+###<a name="section7.1"></a>`GET /v1/admin/metrics`
+Gets metrics for a whole Cygnus agent. Specifically:
+
+* `incomingTransactions`. Number of incoming transactions (a transaction involves a request and a response).
+* `incomingTransactionRequestSize`. Total size of the requests, in bytes.
+* `incomingTransactionResponseSize`. Total size of the responses, in bytes.
+* `incomingTransactionError`. Number of transactions causing an error.
+* `serviceTime`. Average time between transaction requests reception and transaction responses sending.
+
+Metrics are only gathered if the following custom Cygnus components are used:
+
+* `NGISRestHandler`
+* Any sink extending `NGSISink`.
+
+```
+GET http://<cygnus_host>:<management_port>/v1/admin/metrics[?reset=true|false]
+```
+
+Response:
+
+```
+200 OK
+
+{
+    "services": {
+        "service1": {
+            "subservs": {
+                "/subservice1": {
+                    <metrics for subservice1 within service1>
+                },
+                "/subservice1": {
+                    <metrics for subservice2 within service1>
+                }
+            },
+            "sum": {
+                <aggregated metrics for all subservices within service1>
+            }
+        },
+        "service2": {
+            "subservs": {
+                "/subservice1": {
+                    <metrics for subservice1 within service2>
+                },
+                "/subservice2": {
+                    <metrics for subservice2 within service2>
+                }
+            },
+            "sum": {
+                <aggregated metrics for all subservices within service2>
+            }
+        }
+    },
+    "sum": {
+        "subservs": {
+            "/subservice1": {
+                <aggregated metrics for subservice1 within all the services>
+            },
+            "/subservice2": {
+                <aggregated metrics for subservice2 within all the services>
+            }
+        },
+        "sum": {
+            <aggregated metrics for all subservices within all the services>
+        }
+    }
+}
+```
+
+If `reset=true` then metrics and returned and immediatelly after they are deleted (gathering the metrics and deleting them is an atomic operation, i.e. another interleaved GET operation will wait until the deletion is done).
+
+Additionally, because Cygnus distributes event processing among sources (responsible for event reception) and sinks (responsible for event persistence; an event may be processed by 2 or more sinks in parallel), some considerations when retrieving metrics must be had into account:
+
+* The number of incoming transactions may be inconsistent with regards to the service time, because certain events may be in a reception state, but not in a processed state.
```

Comment:
inconsistent -> eventually inconsistent ?

To mean that when times passes all becomes consisent again